flink-streaming-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09Fetcher.java

```diff
@@ -143,133 +123,26 @@ public Kafka09Fetcher(
 
 	@Override
 	public void runFetchLoop() throws Exception {
```

Comment:
Ok, I agree to be safe.

Also, I just realized that "end of stream" shouldn't lead to the `ClosedException`, only "cancellation", "fetcher error", "consumer error", and (hopefully not) any other stuff we overlooked will. So, basically, like what you said, only abnormal terminations. In that case, let's keep it this way.